### ISSUE TITLE
Harden all endowments

### ISSUE THREAD

**Issue by rekmarks** [MetaMask/snaps-monorepo#585]
To the greatest extent possible, endowment made available to a snap should be hardened via SES's `harden()` function. This is to preclude the existence of sandbox escapes by e.g. walking properties to obtain a reference to the root realm `globalThis`.

This effort is complicated by the existence of endowments that may depend on mutable properties, for example the `wallet` provider object.

For all of the endowments we pass to snaps, we should:
1. Attempt to harden the endowment by passing it to `harden()`.
2. If that doesn't work, ensure that its prototype has been passed to `harden()`, and take whatever other measures may be necessary to close unsanctioned communications channels with the root realm.

**Comment by rekmarks** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Snaps-Platform-615b3a7c08d2b20015eb6c4e/issues/MetaMask/snaps-monorepo/585?planning-poker) with Zenhub @david0xd @FrederikBolding @GuillaumeRx @hmalik88 @Mrtenz @ritave

**Comment by kenhkan** [comment]
Note from standup: This should be done prior to audit from Least Authority. Date to be determined.

**Comment by david0xd** [comment]
It is found that the Jest test framework does not work with SES. That makes impossible to test code that includes SES functions such as `harden({...})`. Replacement of the test framework is needed in the `snaps-execution-environments` package in order to continue work on this task properly.

Research of the potential test framework that can be used with SES is followed up in another ticket here: https://github.com/MetaMask/snaps-monorepo/issues/1003
